src/main/java/com/bart/scorebetlive442/service/LeagueService.java

```diff
@@ -95,23 +93,87 @@ public void addTeamToLeague(Long leagueId, Set<Long> teamIds) {
         var existingLeagueOpt = leagueRepository.findById(leagueId);
 
         if (existingLeagueOpt.isEmpty()) {
-            throw new RuntimeException("nie-ma-ligi");
+            throw new RuntimeException("No such league");
         }
 
-        List<TeamEntity> teamsToAdd = new ArrayList<>();
+        List<TeamEntity> foundTeamsById= teamRepository.findAllById(teamIds);
 
-        var allById = teamRepository.findAllById(teamIds);
+        Set<Long> foundTeamsIds = foundTeamsById
+                .stream()
+                .map(TeamEntity::getId)
+                .collect(Collectors.toSet());
+        Set<Long> missingTeamIds = new HashSet<>(teamIds);;
+        missingTeamIds.removeAll(foundTeamsIds);
 
-        if (allById.size() != teamIds.size()) {
-            // praca domowa: jak sprawdzic której drużyny nie ma
-            throw new RuntimeException("nie ma druzyny");
+        if (!missingTeamIds.isEmpty()) {
+            throw new RuntimeException("Teams not found" + missingTeamIds);
```

Comment:
koniec końców ta logika jest dobra ale zbyt skoplikowana - i ciężka do przeczytania :) a można prościej:
(nie pamiętam jak omawialiśmy to na zajęciach ale finalnie musimy rozważyć, obsłużyć takie przypadki):

tutaj przydaje się wiedza z ISTQB :), jak rozpatrzyć wszystkie najważniejsze przypadki - możemy do tego wrócić na zajęciach:

![image](https://github.com/user-attachments/assets/f830c82c-a0f7-4d2c-b0d3-9bc712a21e34)

Reply:
omówimy to na zajęciach